### ISSUE TITLE
Warning about useless lifetime

### ISSUE THREAD

**Issue by antoyo** [Manishearth/rust-clippy#459]
Hello.
The following code does not give any warning from `clippy`:

```rust
fn test<'a>(a: i32) -> i32 {
    a
}

fn main() {
    test(10);
}
```

but I think it should, because the lifetime is not used.

Thanks to add this lint.
